### PR TITLE
🕙 Expire old ivr runs

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

**Comment by norkans7** [comment]
Addresses https://github.com/rapidpro/rapidpro/issues/815

File: temba/flows/models.py [hunk 1]
```diff
@@ -2097,6 +2100,10 @@ def start_call_flow(self, all_contact_ids, start_msg=None, extra=None, flow_star
             run.connection = call
             run.save(update_fields=["connection"])
 
+            # update our expiration date on our run initially to 7 days for IVR, that will be adjusted when the call is answered
+            next_week = timezone.now() + timedelta(days=7)
```
Comment (dodobas) [inline temba/flows/models.py]: should `7 days` be a constant ?

File: temba/ivr/models.py [hunk 2]
```diff
@@ -219,9 +219,10 @@ def update_status(self, status: str, duration: float, channel_type: str):
             self.IN_PROGRESS,
             self.RINGING,
         ):
-            runs = FlowRun.objects.filter(connection=self, is_active=True, expires_on=None)
+            runs = FlowRun.objects.filter(connection=self, is_active=True)
             for run in runs:
-                run.update_expiration()
+                if not run.expires_on or (run.expires_on - run.created_on > timedelta(days=2)):
```
Comment (dodobas) [inline temba/ivr/models.py]: should `2 days` be a constant ?

File: temba/flows/migrations/0187_fix_expiration_date_for_ivr_and_ussd_runs.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            connection = run.connection.get()
+            connection.close()
+
+            run.is_active = False
+            run.exited_on = now
+            run.exit_type = "E"
+            run.modified_on = now
+            run.child_context = None
+            run.parent_context = None
+            run.save()
```
Comment (dodobas) [inline temba/flows/migrations/0187_fix_expiration_date_for_ivr_and_ussd_runs.py]: maybe we could use `save(update_fields=...)`

File: temba/ivr/models.py (hunk 2 above)
Comment (norkans7) [reply temba/ivr/models.py]: Use the last/biggest option in the available choices

File: temba/flows/views.py [hunk 4]
```diff
@@ -89,19 +89,6 @@
 )
 
 
-IVR_EXPIRES_CHOICES = (
```
Comment (norkans7) [inline temba/flows/views.py]: moved to IVRCall

**Pull request merged**